examples/02_sphere/main.rs

```diff
@@ -1,95 +1,51 @@
-//! Displays a multicolored sphere to the user.
-//!
-//! TODO: Rewrite for new renderer.
+//! Displays a shaded sphere to the user.
 
 extern crate amethyst;
 extern crate cgmath;
 extern crate futures;
 extern crate genmesh;
 
 use amethyst::assets::{AssetFuture, BoxedErr};
-use amethyst::ecs::rendering::{Factory, LightComponent, MaterialComponent, MeshComponent,
-                               RenderBundle};
+use amethyst::ecs::World;
+use amethyst::ecs::rendering::{AmbientColor, Factory, LightComponent, MaterialComponent,
+                               MeshComponent, RenderBundle};
 use amethyst::ecs::transform::Transform;
 use amethyst::prelude::*;
-use amethyst::renderer::Config as DisplayConfig;
+use amethyst::renderer::{Config as DisplayConfig, Rgba};
 use amethyst::renderer::prelude::*;
 use cgmath::{Deg, Vector3};
 use cgmath::prelude::InnerSpace;
 use futures::{Future, IntoFuture};
 use genmesh::{MapToVertices, Triangulate, Vertices};
 use genmesh::generators::SphereUV;
 
+const SPHERE_COLOUR: Rgba = Rgba(0.0, 0.0, 1.0, 1.0); // blue
+const AMBIENT_LIGHT_COLOUR: Rgba = Rgba(0.01, 0.01, 0.01, 1.0); // near-black
+const POINT_LIGHT_COLOUR: Rgba = Rgba(1.0, 1.0, 1.0, 1.0); // white
+const BACKGROUND_COLOUR: [f32; 4] = [0.0, 0.0, 0.0, 0.0]; // black
+const LIGHT_POSITION: [f32; 3] = [2.0, 2.0, -2.0];
+const LIGHT_RADIUS: f32 = 5.0;
+const LIGHT_INTENSITY: f32 = 3.0;
+
 struct Example;
 
-fn load_proc_asset<T, F>(engine: &mut Engine, f: F) -> AssetFuture<T::Item>
+fn load_proc_asset<T, F>(world: &mut World, f: F) -> AssetFuture<T::Item>
```

Comment:
We should really replace this with the new asset management methods, which I think I've already added.

Reply:
All examples use these functions. I think it might be better to solve them all in a separate PR after this and the pong PR merged.

Reply:
As long as we can get rid of creating this function in every example, I'm fine with that.